src/main/java/uk/gov/beis/els/api/openapi/OpenApiPropertyCustomiser.java

```diff
@@ -72,13 +73,28 @@ private String stringifyDigitsConstraint(Optional<Digits> digits) {
     }
     int ints = digits.get().integer();
     int fractions = digits.get().fraction();
-    if (ints > 0 && fractions > 0) {
-      return String.format(". This may be up to %d digit(s) long with an optional %d decimal places.", ints, fractions);
+
+    String intsString;
+    String fractionsString;
+    if (ints > 1) {
+      intsString = String.format(". This may be up to %d digits long", ints);
     } else if (ints > 0) {
-      return String.format(". This may be up to %d digit(s) long.", ints);
+      intsString = ". This may be up to 1 digit long";
     } else {
-      return "";
+      intsString = "";
+    }
+
+    if (fractions > 1) {
+      fractionsString = String.format(" with an optional %d decimal places.", fractions);
+    } else if (fractions > 0) {
+      fractionsString = " with an optional decimal place.";
+    } else if (!StringUtils.isBlank(intsString)){
+      fractionsString = ".";
+    } else {
+      fractionsString = "";
     }
+
+    return intsString + fractionsString;
```

Comment:
Nit; I think this would be a bit easier to read and result in less duplication if the 'pluralising' logic was split into its own function. e.g.
```
private String pluralise(int count, String item) {
  if (count == 1) {
    return item;
  } else {
    return item + "s";
  }
}
```

Then invoked like
```
if (ints > 0 && fractions > 0) {
  return String.format(". This may be up to %d %s long with an optional %d decimal %s.",
      ints,
      pluralise(ints, "digit"),
      fractions,
      pluralise(fractions, "place")
  );
} else if (ints > 0) {
  return String.format(". This may be up to %d %s long.", ints, pluralise(ints, "digit"));
} else {
  return "";
}
```

Reply:
done